src/agent/agent.c

```diff
@@ -535,6 +530,20 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
                 return false;
         }
 
+        result = cfg_set_default_section(agent->config, CFG_SECT_AGENT);
```

Comment:
Any reason not to move `cfg_set_default_section` invocation into `cfg_agent_def_conf`?